### PR TITLE
Convert relative paths to absolute as their associated sls file is processed

### LINKED ISSUE #54179: Include issue with absolute and relative path result in conflicting IDs
### Description of Issue
I think include does not work as expected when include relative and absolute path

### Setup

- File **salt/bug/init.sls** :

```yaml
include:
  - .bug
  - .include
```

- File  **salt/bug/bug.sls**:

```yaml
bug:
  test.succeed_without_changes:
    - name: bug
```

- File  **salt/bug/include.sls**:
```yaml
include:
  - salt/bug/bug

include_bug:
  test.succeed_without_changes:
    - name: include_bug
    - require:
      - id: bug
```

### Steps to Reproduce Issue
```bash
$ salt-call state.apply salt/bug
local:
    Data failed to compile:
----------
    Detected conflicting IDs, SLS IDs need to be globally unique.
    The conflicting ID is 'bug' and is found in SLS 'base:salt/bug.bug' and SLS 'base:salt/bug/bug'
```

Now if I replace with full path in **salt/bug/init.sls**:
```yaml
include:
  - salt/bug/bug
  - salt/bug/include
```

```bash
$ salt-call state.apply salt/bug
local:
----------
          ID: bug
    Function: test.succeed_without_changes
      Result: True
     Comment: Success!
     Started: 10:20:43.032602
    Duration: 0.0 ms
     Changes:
----------
          ID: include_bug
    Function: test.succeed_without_changes
      Result: True
     Comment: Success!
     Started: 10:20:43.033596
    Duration: 0.0 ms
     Changes:

Summary for local
------------
Succeeded: 2
Failed:    0
------------
```

### Versions Report
```
Salt Version:
           Salt: 2019.2.0

Dependency Versions:
           cffi: 1.11.5
       cherrypy: 17.4.1
       dateutil: 2.7.5
      docker-py: Not Installed
          gitdb: 2.0.5
      gitpython: 2.1.10
          ioflo: Not Installed
         Jinja2: 2.10
        libgit2: 0.28.2
        libnacl: 1.6.1
       M2Crypto: Not Installed
           Mako: 1.0.7
   msgpack-pure: Not Installed
 msgpack-python: 0.5.6
   mysql-python: Not Installed
      pycparser: 2.17
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: 0.28.2
         Python: 3.5.4 (v3.5.4:3f56838, Aug  8 2017, 02:17:05) [MSC v.1900 64 bit (AMD64)]
   python-gnupg: 0.4.3
         PyYAML: 3.13
          PyZMQ: 17.1.2
           RAET: Not Installed
          smmap: 2.0.5
        timelib: 0.2.4
        Tornado: 4.5.3
            ZMQ: 4.2.5

System Versions:
           dist:
         locale: cp1252
        machine: AMD64
        release: 10
         system: Windows
        version: 10 10.0.18362 SP0 Multiprocessor Free
```

### PR DESCRIPTION
**Pull request by dmurphy18**
### What does this PR do?
Converts the relative path sls files to their equivalent absolute path when initially rendering their state.

### What issues does this PR fix or reference?
Fixes: https://github.com/saltstack/salt/issues/54179

### Previous Behavior
Would get conflicting ID's for the same sls file when rendering the paths for relative and absolute paths to the same file. For example:

salt/bug.bug
salt/bug/bug

are the same sls file from the perspective of the salt/bug directory, but would generate conflicting ids

### New Behavior
When rendering the relative path salt/bug.bug, it is now converted to it's absolute path salt/bug/bug.
This is similar to converting all local time to UTC such that comparisons are simplified, likewise using absolute paths.

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [X] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [X] Tests written/updated

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: tests/pytests/integration/states/test_highstate_paths.py [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+    tmp_path,
+    salt_minion,
+):
+    """
+    This tests for any regressions for this issue:
+    https://github.com/saltstack/salt/issues/54179
+    """
+    test_tempdir = salt_master.state_tree.base.paths[0] / "tmp_dir"
+    test_tempdir.mkdir(parents=True, exist_ok=True)
+
```
Comment (s0undt3ch) [inline tests/pytests/integration/states/test_highstate_paths.py:65]:
```suggestion
```
The directory should automatically be created. It didn't for you?

File: tests/pytests/integration/states/test_highstate_paths.py [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
```
Comment (s0undt3ch) [inline tests/pytests/integration/states/test_highstate_paths.py:193]:
```suggestion
```
Same as above

File: tests/pytests/integration/states/test_highstate_paths.py [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
```
Comment (s0undt3ch) [inline tests/pytests/integration/states/test_highstate_paths.py:257]:
```suggestion
```

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

File: tests/pytests/integration/states/test_highstate_paths.py (hunk 1 above)
Comment (s0undt3ch) [reply tests/pytests/integration/states/test_highstate_paths.py:65]: https://github.com/saltstack/pytest-salt-factories/blob/master/src/saltfactories/utils/tempfiles.py#L147

Comment (dmurphy18) [reply tests/pytests/integration/states/test_highstate_paths.py]: done

File: tests/pytests/integration/states/test_highstate_paths.py (hunk 2 above)
Comment (dmurphy18) [reply tests/pytests/integration/states/test_highstate_paths.py]: Done

File: tests/pytests/integration/states/test_highstate_paths.py [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+    pillar_tree,
+    tmp_path,
+    salt_minion,
+):
+    """
+    This tests for any regressions for this issue:
+    https://github.com/saltstack/salt/issues/54179
+    """
+    ## test_tempdir = salt_master.state_tree.base.paths[0] / "tmp_dir"
+
```
Comment (s0undt3ch) [inline tests/pytests/integration/states/test_highstate_paths.py:64]:
```suggestion

```
These can now go?

**Review by s0undt3ch — DISMISSED** [review]

**Review by twangboy — DISMISSED** [review]

**Review by waynew — DISMISSED** [review]
Well, the only thing that I was particularly worried about was something like the IDs changing - it *does* change `__sls__` if you're using `--out=json`, but aside from that, based on some internal discussion/exploration it doesn't seem to be a problem, and all of the tests are passing, especially saltcheck which looks like one of the main places that `__sls__` is being used.

I don't have any reason to block this, and I've tested it locally and it appears to do what it says on the tin :+1:

**Comment by dmurphy18** [comment]
re-run fedora
re-run windows

**Comment by dmurphy18** [comment]
re-run freebsd

**Review by garethgreenaway — APPROVED** [review]

**Review by twangboy — APPROVED** [review]

**Review by waynew — APPROVED** [review]
:rocket:

**Pull request merged**